nixosModules/homestakeros/default.nix

```diff
@@ -237,26 +233,54 @@ in
               '';
             in
             {
-              description = "Start the SSV node if the private operator key exists";
-              unitConfig.ConditionPathExists = [
-                "${cfg.addons.ssv-node.privateKeyFile}"
-                "${cfg.addons.ssv-node.privateKeyPasswordFile}"
-              ];
-              # The operator key is defined here, so it does not need to be evaluated
-              script = ''
-                ${pkgs.ssvnode}/bin/ssvnode start-node --config ${ssvConfig}
-              '';
+              description = "Operator node for Secret Shared Validators (SSV)";
+              after = [ "network-online.target" ];
+              wants = [ "network-online.target" ];
               wantedBy = [ "multi-user.target" ];
+
+              script =
+                let
+                  curl = "${pkgs.curl}/bin/curl";
+                  jq = "${pkgs.jq}/bin/jq";
+                in
+                ''
+                  mkdir -p ${cfg.addons.ssv-node.dataDir}
+
+                  # Check if keys exist
+                  if [ ! -f "${cfg.addons.ssv-node.privateKeyFile}" ] || [ ! -f "${cfg.addons.ssv-node.publicKeyFile}" ]; then
+
+                    # Check if dataDir is on tmpfs
+                    FS_TYPE=$(df --output=fstype "${cfg.addons.ssv-node.dataDir}" | tail -n1)
+                    if [[ "$FS_TYPE" =~ ^(tmpfs|overlay|squashfs)$ ]]; then
+                      echo "error: '${cfg.addons.ssv-node.dataDir}' is on a tmpfs; generated keys would be lost after reboot"
+                      exit 1
+                    fi
```

Comment:
Notice that you could have just use the `cfg` to track down the `dataDir` to the corresponding `systemd-mount` option hence get static guarantees rather than do this runtime checking.